lib/ui/widgets/trailing_icon_row.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:zachranobed/common/constants.dart';
+
+/// A stateless widget that represents a row with trailing info and icon.
+///
+/// This row includes a title, description, trailing information, and an icon.
+/// It is designed to be clickable, triggering a callback function when tapped.
+class TrailingIconRow extends StatelessWidget {
+  /// The main title text displayed in the row.
+  final String title;
+
+  /// The description text displayed in the row.
+  final String description;
+
+  /// The text displayed next to the icon at the trailing end.
+  final String trailInfo;
+
+  /// The icon displayed at the trailing end.
+  final IconData trailingIcon;
+
+  /// The function to be called when the row is tapped.
+  final VoidCallback onTap;
+
+  const TrailingIconRow({
+    super.key,
+    required this.title,
+    required this.description,
+    required this.trailInfo,
+    required this.trailingIcon,
+    required this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+        padding: const EdgeInsets.only(bottom: 10.0),
+        child: InkWell(
+          onTap: onTap,
+          child: Container(
+            decoration: BoxDecoration(
+              border: Border.all(
+                color: ZOColors.borderColor,
+                width: 1,
+              ),
+              borderRadius: BorderRadius.circular(10),
+            ),
+            child: ListTile(
+                title: Text(title,
+                    style: const TextStyle(fontSize: FontSize.s),
+                    maxLines: 1,
+                    overflow: TextOverflow.ellipsis),
```

Comment:
Nit: It is hard to readable formatting 🙈

I would add some trailing commas and reformat the file. If you don't mind I will push a commit with a change to show what I mean